pkg/hostagent/status.go

```diff
@@ -78,11 +79,13 @@ func (agent *HostAgent) RunStatus() {
 			UsedIPs: len(agent.usedIPs),
 		}
 		json.NewEncoder(w).Encode(status)
+		agent.ipamMutex.Unlock()
 		agent.indexMutex.Unlock()
 	})
 	http.HandleFunc("/ipam", func(w http.ResponseWriter, r *http.Request) {
 		w.Header().Set("Content-Type", "application/json")
 		agent.indexMutex.Lock()
+		agent.ipamMutex.Lock()
```

Comment:
What is the relevance of adding a second mutex here? You already have one mutex which you are locking and unlocking at the same place...

Reply:
indexMutex is acquired when we access cache in hostagent pkg (agent.usedIP in this case).
ipamMutex is acquired when we access shared variable in ipam pkg.(agent.podIps.CombineV4())

Reply:
I dont think we would need both locks, it should be only one of them. Since it is same type of lock (sync.Mutex).